core/statix.multilang.eclipse/src/main/java/mb/statix/multilang/eclipse/ConfigResourceChangeListener.java

```diff
@@ -0,0 +1,91 @@
+package mb.statix.multilang.eclipse;
+
+import mb.log.api.Logger;
+import mb.spoofax.eclipse.SpoofaxPlugin;
+import mb.spoofax.eclipse.util.StatusUtil;
+import mb.statix.multilang.MultiLangAnalysisException;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.SafeRunner;
+import org.eclipse.jface.util.SafeRunnable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@link IResourceChangeListener} that listens to changes to {@code multilang.yaml} files, and notifies all registered
+ * {@link ConfigChangeListener config change listeners}.
+ */
+public class ConfigResourceChangeListener implements IResourceChangeListener {
```

Comment:
I'm a bit confused as to why this is needed. Didn't we agree that all config file parsing should be done in PIE tasks, so that PIE does the change tracking for us?